### ISSUE TITLE
Generate two-factor authentication (2FA) recovery codes

### ISSUE THREAD

**Issue by CybrZr00** [aspnet/Docs#7019]
For some reason I am unable to generate recovery codes, I followed this tutorial with only one issue (Had to shorten the Site name, might be worth mentioning the max length) until I tested, and tried to get recovery codes. Firstly there were no codes shown, so I click to reset the recovery codes, which looked to have been done, but no where are they displayed! Is this a bug?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 117c950a-0bcb-d936-31cf-5859b4280abb
* Version Independent ID: 270bedc7-a636-5bd0-9e2f-e42235deec8c
* Content: [Enable QR Code generation for authenticator apps in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/identity-enable-qrcodes?view=aspnetcore-2.1)
* Content Source: [aspnetcore/security/authentication/identity-enable-qrcodes.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/security/authentication/identity-enable-qrcodes.md)
* Product: **asp.net-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
@CybrZr00  is this where you don't get the `qrCode`?

```
@section Scripts {
    @await Html.PartialAsync("_ValidationScriptsPartial")

    <script type="text/javascript" src="~/lib/qrcode.js"></script>
    <script type="text/javascript">
        new QRCode(document.getElementById("qrCode"),
            {
                text: "@Html.Raw(Model.AuthenticatorUri)",
                width: 150,
                height: 150
            });
    </script>
}
```

If you, you'll need to debug *qrcode.js* to figure out why it's not connected.

**Comment by CybrZr00** [comment]
Hi Rick, sorry it seems I was a little vague!

The QR code was generated and I connected to the authenticator app successfully, and was able to log in. The issue was with their not being any recovery codes, so if a user were to lose or wipe their phone, they would have no simple way to recover the account. I can see looking at the pages that there should be recovery codes, but for some reason they aren't displayed anywhere. I tried creating new codes, resetting 2fa and the app, and tried logging out and in again with no joy. I noticed some conditional razor code for when the recovery codes drop to <= 3 etc but none of those conditions were met either. I hope that's enough to go on, if not let me know!
